react/src/hooks/index.tsx

```diff
@@ -509,3 +509,45 @@ type BackendAIConfig = {
   debug: boolean;
   [key: string]: any;
 };
+
+export const useAppDownloadMap = () => {
+  const baiClient = useSuspendedBackendaiClient();
+  const appDownloadUrl = baiClient?._config?.appDownloadUrl || '';
+
+  const appDownloadMap = {
+    Linux: {
+      os: 'linux',
+      architecture: ['arm64', 'x64'],
+      extension: 'zip',
+    },
+    MacOS: {
+      os: 'macos',
+      architecture: ['arm64', 'x64'],
+      extension: 'dmg',
+    },
+    Windows: {
+      os: baiClient?.supports('use-win-instead-of-win32') ? 'win' : 'win32',
+      architecture: ['arm64', 'x64'],
+      extension: 'zip',
+    },
+  };
+
+  const [selectedOS, setSelectedOS] =
+    useState<keyof typeof appDownloadMap>('MacOS');
```

Comment:
Instead of keeping this state in `useAppDownloadMap`, it would be better to change `getDownloadLink` to accept `os` and `architecture` as arguments. Additionally, create a separate hook to determine current OS(navigator.platform and userAgent) and set it as the default in `DesktopAppDownloadModal`.

Reply:
Hmm.. does it necessary to make a new hook to determine current OS? I think it can be handled by extra function.